pyanaconda/ui/gui/spokes/installation_source.py

```diff
@@ -462,28 +464,46 @@ def _method_changed(self):
             # to know how to fetch, and the refresh method needs that to know
             # which element of the combo to default to should this spoke be
             # revisited.
-            if self._ftp_active() and not url.startswith("ftp://"):
+            if not url.startswith("ftp://"):
                 url = "ftp://" + url
+            if old_source.method == "url" and not self._proxyChange and \
```

Comment:
Not really needed for this patch series but looking at the code it might make sense to move more stuff constants/enums in the future as the source spoke seems to still use "magical strings" pretty heavily.